### ISSUE TITLE
Feature Proposal: Throttling support for parallel tasks

### ISSUE THREAD

**Issue by nikoskalogridis** [douglascrockford/RQ#13]
Although in an ideal world, the number of executing parallel tasks should not matter in our systems it takes a certain number of concurrent requests before things start to deteriorate.
It would be nice if an additional throttle function is included in the library that will behave similarly to parallel but will limit the number of parallel executing tasks up to a limit parameter.
i.e.
```RQ.throttle(requireds, milliseconds, requiredsConcurrentLimit, optionals, untilliseconds, optionalsConcurrentLimit)```

although the number of arguments is becoming ridiculously big and probably a `params` object with named keys would be a more sensible approach.

I could spent some time to produce a pull request for that if it is agreed that such a feature would be a welcomed addition

**Comment by douglascrockford** [comment]
I don't see the need for it.

**Comment by EvgenyOrekhov** [comment]
@douglascrockford Here is a real world example I just experienced.

There was a need to make a thousand HTTP requests to an external system in parallel, when I tried to make them without a concurrency limit, the external system went down. By trial and error I found a concurrency limit parameter (10) that doesn't crash the external system and doesn't underutilize its resources.

So there very well might be a need for a concurrency limit parameter.
